packages/env-options/src/env-options.js

```diff
@@ -0,0 +1,98 @@
+// @ts-check
+
+// `@endo/env-options` needs to be imported quite early, and so should
+// avoid importing from ses or anything that depends on ses.
+
+// /////////////////////////////////////////////////////////////////////////////
+// Prelude of cheap good - enough imitations of things we'd use or
+// do differently if we could depend on ses
+
+const { freeze } = Object;
+const { apply } = Reflect;
+
+// Should be equivalent to the one in ses' commons.js even though it
+// uses the other technique.
+const uncurryThis =
+  fn =>
+  (receiver, ...args) =>
+    apply(fn, receiver, args);
+const arrayFlatMap = uncurryThis(Array.prototype.flatMap);
+const arrayPush = uncurryThis(Array.prototype.push);
+const arrayJoin = uncurryThis(Array.prototype.join);
+
+const q = JSON.stringify;
+
+const Fail = (template, ...args) => {
+  const msg = arrayFlatMap(args, (arg, i) => [template[i], arg]);
+  arrayPush(msg, template[args.length]);
+  throw Error(arrayJoin(msg, ''));
+};
+
+// end prelude
+// /////////////////////////////////////////////////////////////////////////////
+
+/**
+ * `makeEnvironmentCaptor` provides a mechanism for getting environment
+ * variables, if they are needed, and a way to catalog the names of all
+ * the environment variables that were captured.
+ *
+ * @param {object} aGlobal
+ */
+export const makeEnvironmentCaptor = aGlobal => {
+  const capturedEnvironmentOptionNames = [];
```

Comment:
I think we also want to eagerly capture the `env` object.
```suggestion
  const globalProcess = aGlobal.process;
  const globalEnv = globalProcess && typeof globalProcess === 'object'
    ? globalProcess.env
    : undefined;
  const capturedEnvironmentOptionNames = [];
```

Reply:
See answer below to your "should we instead memoize" question.

Resolving.